mmlearn/tasks/classification.py

```diff
@@ -0,0 +1,250 @@
+"""Zero-shot and linear probing classification evaluation task."""
+
+import inspect
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Literal, Optional, Union
+
+import torch
+import torchmetrics
+from hydra_zen import store
+from pytorch_lightning import LightningModule, Trainer
+from torch.nn import functional
+from torch.utils.data import DataLoader, Dataset
+from torchmetrics import Metric, MetricCollection
+from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+
+from mmlearn.constants import NAME_KEY
+from mmlearn.datasets.core import Modalities
+from mmlearn.datasets.core.data_collator import collate_example_list
+from mmlearn.datasets.core.example import Example
+from mmlearn.datasets.core.modalities import Modality
+from mmlearn.tasks.hooks import EvaluationHooks
+
+
+@dataclass
+class ClassificationTaskSpec:
+    """Specification for a classification task."""
+
+    metric_name: str
+    query_modality: str
+    top_k: List[int]
+
+
+@store(group="eval_task", provider="mmlearn")
+class Classification(EvaluationHooks):
+    """
+    Zero-shot classification evaluation task.
+
+    This task evaluates the zero-shot classification performance.
+
+    Parameters
```

Comment:
Reminder to update this with all the parameters in the `__init__` method.

Reply:
Done